Assets/MRTK/Examples/Experimental/SceneUnderstanding/Scripts/DemoSceneUnderstandingController.cs

```diff
@@ -152,7 +152,10 @@ public void OnObservationRemoved(MixedRealitySpatialAwarenessEventData<SpatialAw
         {
             RemoveFromData(eventData.Id);
 
-            observedSceneObjects[eventData.SpatialObject.SurfaceType].Remove(eventData.Id);
+            foreach (var sceneObjectDict in observedSceneObjects.Values)
```

Comment:
just clarifying here. there are _no_ changes to the collection during enumeration here, correct? if there (possibly) are, we are at risk of an exception,.

Reply:
`observedSceneObjects` is a collection (i.e. dictionary) of dictionaries and I modify the content of individual dictionaries (`sceneObjectDict`) inside that collection. I think this should be fine right?